### ISSUE TITLE
Docker Engine won't start after installation of KB5043145 

### ISSUE THREAD

**Issue by thebrandre** [docker/for-win#14336]
### Description

After installation of KB5043145, the Docker service no longer starts. I get the following log output when starting docker directly with --debug:

> cleanupServiceBindings for ff14bf2799aa103a38f57b5d512c9840dba5d897d5f209729c2ed32c259a1aab
Allocating IPv4 pools for network Default Switch (ff14bf2799aa103a38f57b5d512c9840dba5d897d5f209729c2ed32c259a1aab)
RequestPool: {AddressSpace:LocalDefault Pool:172.23.96.0/20 SubPool: Options:map[] Exclude:[] V6:false}
RequestAddress(172.23.96.0/20, 172.23.96.1, map[RequestAddressType:com.docker.network.gateway])
[GET]=>[/endpoints/] Request : 
Allocating IPv4 pools for network nat (a4bebd03194e16e822a50fb14da64a00e022b87e2beeb39b7348a282139c5ae3)
RequestPool: {AddressSpace:LocalDefault Pool:192.168.254.1/24 SubPool: Options:map[] Exclude:[] V6:false}
RequestAddress(192.168.254.1/24, <nil>, map[RequestAddressType:com.docker.network.gateway])
HNSNetwork Request ={\"Name\":\"nat\",\"Type\":\"nat\",\"Subnets\":[{\"AddressPrefix\":\"192.168.254.1/24\"}]} Address Space=[{192.168.254.1/24  []}]
[POST]=>[/networks/] Request : {\"Name\":\"nat\",\"Type\":\"nat\",\"Subnets\":[{\"AddressPrefix\":\"192.168.254.1/24\"}]}
releasing IPv4 pools from network nat (a4bebd03194e16e822a50fb14da64a00e022b87e2beeb39b7348a282139c5ae3)
ReleasePool(192.168.254.1/24)
daemon configured with a 15 seconds minimum shutdown timeout
start clean shutdown of all containers with a 15 seconds timeout...
failed to start daemon: Error initializing network controller: error creating default network: failed during hnsCallRawResponse: hnsCall failed in Win32: Element not found. (0x490

It works again if I ...
- uninstall KB5043145, uninstall Docker
- remove `C:\ProgramData\docker`,
- remove `D:\docker`
- reboot the host system
- reinstall Docker Engine

Using the following settings in `daemon.json`:
```json
{
  "storage-opts": [
    "size=500GB"
  ],
  "data-root": "d:\\docker",
  "fixed-cidr": "192.168.254.1/24",
  "features": {
    "windows-dns-proxy": false
  }
}
```

### Reproduce

1. Install Windows Update KB5043145 on Windows Professional 23H2, build 22631.4169, Windows Feature Experience Pack 1000.22700.1034.0
2. Reboot system
3. Try to use the Docker service - but it is no running nor does it start when invoked manually

### Expected behavior

The service usually starts successfully. ;-)

### docker version

```bash
Client:
 Version:    27.3.1
 Context:    default
 Debug Mode: false

Server:
 Containers: 0
  Running: 0
  Paused: 0
  Stopped: 0
 Images: 7
 Server Version: 27.3.1
 Storage Driver: windowsfilter
  Windows:
 Logging Driver: json-file
 Plugins:
  Volume: local
  Network: ics internal l2bridge l2tunnel nat null overlay private transparent
  Log: awslogs etwlogs fluentd gcplogs gelf json-file local splunk syslog
 Swarm: inactive
 Default Isolation: hyperv
 Kernel Version: 10.0 22631 (22621.1.amd64fre.ni_release.220506-1250)
 Operating System: Microsoft Windows Version 23H2 (OS Build 22631.4169)
 OSType: windows
 Architecture: x86_64
 CPUs: 32
 Total Memory: 127.9GiB
 Name: vrmbuild001
 ID: 3e3d2a09-4bd4-42df-ab22-cd2ebec91bef
 Docker Root Dir: d:\docker
 Debug Mode: false
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
 Product License: Community Engine
```

### docker info

```bash
Client:
 Version:    27.3.1
 Context:    default
 Debug Mode: false

Server:
 Containers: 0
  Running: 0
  Paused: 0
  Stopped: 0
 Images: 7
 Server Version: 27.3.1
 Storage Driver: windowsfilter
  Windows:
 Logging Driver: json-file
 Plugins:
  Volume: local
  Network: ics internal l2bridge l2tunnel nat null overlay private transparent
  Log: awslogs etwlogs fluentd gcplogs gelf json-file local splunk syslog
 Swarm: inactive
 Default Isolation: hyperv
 Kernel Version: 10.0 22631 (22621.1.amd64fre.ni_release.220506-1250)
 Operating System: Microsoft Windows Version 23H2 (OS Build 22631.4169)
 OSType: windows
 Architecture: x86_64
 CPUs: 32
 Total Memory: 127.9GiB
 Docker Root Dir: d:\docker
 Debug Mode: false
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
 Product License: Community Engine
```

### Diagnostics ID

NA (only Docker Engine installed)

### Additional Info

I am using only Docker Engine 27.3.1 but experienced the same issue with 27.1.3.

Host: AMD Ryzen 9 5950X 16-Core Processor 3.40 GHz

**Comment by branpurn** [comment]
Same issue. I uninstalled KB5043145 and Docker is working again, I did not have to reinstall Docker. 

This appears to be an issue with the KB breaking WSL: https://github.com/microsoft/WSL/issues/12087

**Comment by thebrandre** [comment]
Turns out I made a wrong diagnosis here. Sorry about that!

My fix was to uninstall the update and to reinstall Docker (and also delete `daemon.json`).
This just happened to solve the primary problem because `"fixed-cidr"` in `daemon.json` takes effect only after [host system reboot](https://github.com/docker/for-win/issues/13030#issuecomment-1289441950).
So another reboot would also reintroduce the issue independent of the mentioned Windows update.

I am not yet sure why `"fixed-cidr"` does no longer work but that's another story.

**Comment by pantharshit007** [comment]
> Turns out I made a wrong diagnosis here. Sorry about that!
> 
> My fix was to uninstall the update and to reinstall Docker (and also delete `daemon.json`). This just happened to solve the primary problem because `"fixed-cidr"` in `daemon.json` takes effect only after [host system reboot](https://github.com/docker/for-win/issues/13030#issuecomment-1289441950). So another reboot would also reintroduce the issue independent of the mentioned Windows update.
> 
> I am not yet sure why `"fixed-cidr"` does no longer work but that's another story.

what error does `fixed-cidr` and `daemon.json` introduce since I am still not able to run docker and wsl since the last window update properly?
